### PR TITLE
Rewrite UserTable to tsx format and fix the sonar issues

### PR DESCRIPTION
**Pull request by BohdanMylyi**
Rewrite UserTable to tsx format and fix the sonar issues

### CODE REVIEW

**Review by dudchakk — APPROVED** [review]

**Review by YaroslavChuiko — COMMENTED** [review]
Try to fix these SonarCloud issues

image

File: src/components/user-table/UserTable.tsx [hunk 1]
```diff
@@ -13,23 +13,51 @@ import usePagination from '~/hooks/table/use-pagination'
 import Tab from '~/components/tab/Tab'
 import EnhancedTableToolbar from '~/components/enhanced-table/enhanced-table-toolbar/EnhancedTableToolbar'
 import EnhancedTablePagination from '~/components/enhanced-table/enhanced-table-pagination/EnhancedTablePagination'
+import { Sort } from '~/hooks/table/use-sort'
 
 import { styles } from '~/components/user-table/UserTable.styles'
 import { VisibilityEnum } from '~/types'
+import { GetUsersParams } from '~/types'
```
Comment (abalanovsky) [inline src/components/user-table/UserTable.tsx:20]: fix imports please

File: src/hooks/table/use-sort.tsx [hunk 2]
```diff
@@ -36,3 +36,4 @@ const useSort = ({ initialSort }: UseSortProps): SortHook => {
 }
 
 export default useSort
+export type { Sort }
```
Comment (abalanovsky) [inline src/hooks/table/use-sort.tsx:39]: Why are you re-exporting type here?

**Review by abalanovsky — CHANGES_REQUESTED** [review]

**Review by YaroslavChuiko — APPROVED** [review]

File: src/components/user-table/UserTable.tsx [hunk 3]
```diff
@@ -157,6 +190,17 @@ const UserTable = ({
   const toolbarVisibility =
     selected.length > 0 ? VisibilityEnum.Visible : VisibilityEnum.Hidden
 
+  const handleChangePage = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent> | null,
+    page: number
+  ) => {
+    console.log(page)
+  }
```
Comment (Pasichnichenkoilya) [inline src/components/user-table/UserTable.tsx:198]: Is this a test `console.log` or it's just mocked?

Comment (BohdanMylyi) [reply src/components/user-table/UserTable.tsx:198]: yes, just mocked

**Review by docia — APPROVED** [review]

**Pull request merged**